### ISSUE TITLE
dotnet store is too tolerant of package downgrades

### ISSUE THREAD

**Issue by rynowak** [dotnet/sdk#1184]
I had a case recently in https://github.com/aspnet/JitBench/tree/dev where the version of the of the ASP.NET libraries in my manifest file https://github.com/aspnet/JitBench/blob/dev/CreateStore.proj wasn't getting set.

The result is `dotnet store` got all of the `1.0.0` versions and of the packages and printed warnings like:
```
C:\Users\rynowak\.dotnet\x64\sdk\2.0.0-preview1-005899\NuGet.targets(97,5): warning : Dependency specified was Microsoft
.AspNetCore.Authentication.Cookies  but ended up with Microsoft.AspNetCore.Authentication.Cookies 1.0.0. [C:\Users\rynow
ak\git\aspnet\JitBench\.temp\Microsoft.AspNetCore.Authentication.Cookies_\Package.csproj]
```

The operation completed successfully.

It doesn't see useful to allow downgrades when building a store unless I'm missing something obvious...

**Comment by ramarag** [comment]
@rynowak dotnet store does not do any checks. We in fact restore each entry in the manifest individually so that other entries do not upgrade or downgrade each other.
What you are seeing seems to be the default behavior of Nuget. If you get any upgrades in other projects it is because the project is restored as a unit and other referred packages upgrades the aspnet package according ti Nuget Rules

makes sense ?

**Comment by rynowak** [comment]
> makes sense ?

OK so I tried this again by setting the version number to something that doesn't exist (2.0.0-preview3-20303). This fails, and that seems right.

So I think the issue that I'm trying to report here only happens when you omit the version number of a package. I ended up with a store full of 1.0.0 packages when none of my package references had a version specified. That might be NuGet's default behavior, I'd rather have an error than that.
